imagersite/imager_images/views.py

```diff
@@ -1,16 +1,189 @@
+from django.core.exceptions import PermissionDenied
+from django.conf import settings
+from .forms import *
+from djgeojson.serializers import Serializer as GeoJSONSerializer
 
 
+import Algorithmia
+import base64
+
+
+def get_faces(path):
+    with open(settings.MEDIA_ROOT + "/" + path, "rb") as img:
+        bimage = base64.b64encode(img.read())
+    Algorithmia.apiKey = "Simple simivSeptsC+ZsLks5ia0wXmFbC1"
+    result = Algorithmia.algo('/ANaimi/FaceDetection').pipe(bimage)
+    faces = []
+    for rect in result:
+        face = Face()
+        face.name = "Petter Rabbit"
+        face.x = rect['x']
+        face.y = rect['y']
+        face.width = rect['width']
+        face.height = rect['height']
+        faces.append(face)
+        for face in faces:
+            face.save()
+    return faces
+
+
+def set_faces(request, photo_id):
+    photo = Photo.objects.get(pk=photo_id)
+    face_id = request.POST.get('photo_id', '0')
+    face = Face.objects.get(id=face_id)
+    face.name = Face.objects.get('name', 'Unknown')
+    face.save()
+
+
+@login_required
 def photo_view(request, photo_id):
-    photo = Photo.objects.filter(id=photo_id).first()
-    return render(request, 'photo.html', {'photo': photo})
+    photo = get_object_or_404(Photo, pk=photo_id)
+    if request.user != photo.owner:
+        raise PermissionDenied
+    if request.method == 'POST':
+        faces = get_faces(str(photo.file))
+        # set_faces(request, photo_id)
+        return render(request, 'photo.html', {'photo': photo, 'faces': faces})
+    else:
+        return render(request, 'photo.html', {'photo': photo})
 
 
+@login_required
 def album_view(request, album_id):
-    album = Album.objects.filter(id=album_id).first()
+    album = get_object_or_404(Album, pk=album_id)
+    if request.user != album.owner:
+        raise PermissionDenied
     return render(request, 'album.html', {'album': album})
 
 
+@login_required
 def library_view(request):
     return render(request, 'library.html')
+
+
+@login_required
+def add_album_view(request):
+    if request.method == 'POST':
+        form = AlbumForm(request.POST, request.FILES)
+        form.fields['photos'].queryset = Photo.objects.filter(owner=request.user)
+        form.fields['cover'].queryset = Photo.objects.filter(owner=request.user)
+        if form.is_valid():
+            new_album = form.save(commit=False)
+            new_album.owner = request.user
+            new_album.save()
+            form.save_m2m()
+            return HttpResponseRedirect('/images/library')
+    form = AlbumForm()
+    form.fields['photos'].queryset = Photo.objects.filter(owner=request.user)
+    form.fields['cover'].queryset = Photo.objects.filter(owner=request.user)
+    return render(request, 'add_album.html', {'form': form.as_p})
+
+
+@login_required
+def add_photo_view(request):
+    if request.method == 'POST':
+        picform1 = PhotoForm(request.POST, request.FILES)
+        picform2 = LocationForm(request.POST)
```

Comment:
It should be possible to get a single form that does both of these jobs.  That would simplify your work here a great deal.  This is a valid approach, but it does mean more work for you.